lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -650,6 +760,25 @@ export abstract class ClientApplication {
     getConfiguration(): BrowserConfiguration {
         return this.config;
     }
+    
+    /**
+     * Generates a correlation id for a request if none is provided.
+     *
+     * @protected
+     * @param {?Partial<BaseAuthRequest>} [request]
+     * @returns {string}
+     */
+    protected getRequestCorrelationId(request?: Partial<BaseAuthRequest>): string {
+        if (request?.correlationId) {
+            return request.correlationId;
+        }
+
+        if (this.isBrowserEnvironment) {
+            return this.browserCrypto.createNewGuid();
+        }
+
+        return "";
```

Comment:
Just curious, when do we expect this case i.e when the correlation id is an empty string?

Reply:
Okay, got it.. thanks

Reply:
For hybrid code?

Reply:
No, we'll only reach this line if the above two checks dont pass (correlation id isn't provided by the application, and we aren't in browser environment). I have this function returning a blank string so that I don't have to mark it as returning null/undefined. I'll add a comment indicating as such.